### ISSUE TITLE
Seg fault on call to from_pandas() with mode='append'

### ISSUE THREAD

**Issue by bkmartinjr** [TileDB-Inc/TileDB-Py#1086]
```
$ ipython
Python 3.9.7 (default, Sep 10 2021, 14:59:43) 
Type 'copyright', 'credits' or 'license' for more information
IPython 8.2.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import pandas as pd
   ...: import tiledb
   ...: 
   ...: print(tiledb.__version__)
   ...: df=pd.DataFrame(index=['a', 'b'])
   ...: tiledb.from_pandas(uri="/tmp/oops", dataframe=df)
   ...: tiledb.from_pandas(uri="/tmp/oops", dataframe=pd.DataFrame(index=pd.Index(data=[])), mode='append')
   ...: 
0.14.4
Segmentation fault (core dumped)
```

**Comment by Shelnutt2** [comment]
Backtrace:

```
[Switching to Thread 0x7fffb23c5640 (LWP 19855)]
0x00007ffff4651a60 in tiledb::sm::WriterTile::final_size (
    this=0xb0, size=0)
    at /home/seth/git/TileDB/tiledb/../tiledb/sm/tile/writer_ti
le.h:131
131         size_ = size;
(gdb) bt
#0  0x00007ffff4651a60 in tiledb::sm::WriterTile::final_size (
    this=0xb0, size=0)
    at /home/seth/git/TileDB/tiledb/../tiledb/sm/tile/writer_tile.h:131
#1  0x00007ffff48671ae in tiledb::sm::UnorderedWriter::prepare_
tiles_var (this=0x5555565d1c60, name=..., cell_pos=...,
    coord_dups=..., tiles=0x555556251da8)
    at /home/seth/git/TileDB/tiledb/sm/query/unordered_writer.c
c:561
#2  0x00007ffff4865c06 in tiledb::sm::UnorderedWriter::prepare_
tiles (this=0x5555565d1c60, name=..., cell_pos=...,
    coord_dups=..., tiles=0x555556251da8)
    at /home/seth/git/TileDB/tiledb/sm/query/unordered_writer.cc:362
#3  0x00007ffff486574b in operator() (
    __closure=0x7fffffffa180, i=0)
    at /home/seth/git/TileDB/tiledb/sm/query/unordered_writer.c
c:345
#4  0x00007ffff4869803 in operator() (this=0x5555568479f0,
    subrange_start=0, subrange_end=1)
    at /home/seth/git/TileDB/tiledb/../tiledb/sm/misc/parallel_
functions.h:194
#5  0x00007ffff486ab2f```

**Comment by ihnorton** [comment]
Fixed in 0.14.5:
```
In [1]: import pandas as pd

In [2]: import tiledb

In [3]: df=pd.DataFrame(index=['a', 'b'])

In [4]: tiledb.from_pandas(uri="/tmp/oops", dataframe=df)

In [5]: tiledb.from_pandas(uri="/tmp/oops", dataframe=pd.DataFrame(index=pd.Index(data=[])), mode='append')

In [6]: tiledb.version(), tiledb.libtiledb.version()
Out[6]: ((0, 14, 5), (2, 8, 3))
```
